pydicom/data/data_manager.py

```diff
@@ -25,63 +135,94 @@ def online_test_file_dummy_paths():
     return dummy_path_map
 
 
-def get_files(base, pattern):
-    """Return all files from a set of sources.
+def get_files(
+    base: Union[str, os.PathLike],
+    pattern: str = "**/*",
+    dtype: int = DataTypes.DATASET
+) -> List[str]:
+    """Return all matching file paths from the available data sources.
+
+    First searches the local *pydicom* data store, then any locally available
+    external sources, and finally the files available in the
+    pydicom/pydicom-data repository.
+
+    .. versionchanged: 2.1
+
+        Added the `dtype` keyword parameter, modified to search locally
+        available external data sources and the pydicom/pydicom-data repository
 
     Parameters
     ----------
     base : str or PathLike
         Base directory to recursively search.
+    pattern : str, optional
+        The pattern to pass to :meth:`Path.glob`, default (``'**/*'``).
+    dtype : int, optional
+        The type of data to search for when using an external source, one of:
 
-    pattern : str
-        A string pattern to filter the files. Default is "*" and it will return
-        all files.
+        * ``0`` - DICOM dataset
+        * ``1`` - Character set file
+        * ``2`` - Palette file
+        * ``3`` - DICOMDIR file
+        * ``4`` - JPEG file
 
     Returns
     -------
     files : list of str
         The list of filenames matched.
     """
+    base = Path(path_from_pathlike(base))
```

Comment:
Out of interest, why not just `Path(base)` here?

Reply:
Good point, I figured the `path_from_pathlike` call was there for a reason, but it doesn't appear to be.